File: server/server_0/im/src/main/java/com/server_0/im/model/ChatModel.java
package com.server_0.im.model;

import com.server_0.dao.entity.im.chat.TImChatImage;
import com.server_0.dao.entity.shop.TShopCommodity;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @ClassName ChatModel
 * @Description 私信数据模型
 * @Author SuperColorMan
 * @Date 2021/2/20 5:16 下午
 * @ModifyDate 2021/2/20 5:16 下午
 * @Version 1.0
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ChatModel {
    /**
     * 发送者用户id
     */
    String userId;

    /**
     * 发送对方用户id
     */
    String byUserId;

    /**
     * 内容类型
     */
    String type;

    /**
     * 内容
     */
    String content;

    /**
     * 图片信息
     */
    TImChatImage tImChatImage;

    /**
     * 商品信息
     */
    TShopCommodity tShopCommodity;

}

File: app/app_0/lib/model/im/ChatModel.dart
///-------------------------
/// des : 私信模型
/// -------------------------
class ChatModel {
  ///发送者用户id
  String _userId;

  ///发送对方用户id
  String _byUserId;

  /// 内容类型
  String _type;

  /// 内容
  String _content;

  ChatModel(this._userId, this._byUserId, this._type, this._content);

  String get content => _content;

  set content(String value) {
    _content = value;
  }

  String get type => _type;

  set type(String value) {
    _type = value;
  }

  String get byUserId => _byUserId;

  set byUserId(String value) {
    _byUserId = value;
  }

  String get userId => _userId;

  set userId(String value) {
    _userId = value;
  }

  @override
  String toString() {
    return '{"userId": "${_userId}", "byUserId": "${_byUserId}", "type": "${_type}", "content": "${_content}"}';
  }
}
